apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexSmartContractService.java

```diff
@@ -0,0 +1,168 @@
+import com.apollocurrency.aplwallet.apl.core.app.KeyStoreService;
+import com.apollocurrency.aplwallet.apl.core.model.WalletKeysInfo;
+import com.apollocurrency.aplwallet.apl.eth.contracts.DexContract;
+import com.apollocurrency.aplwallet.apl.eth.contracts.DexContractImpl;
+import com.apollocurrency.aplwallet.apl.eth.model.EthWalletKey;
+import com.apollocurrency.aplwallet.apl.eth.service.EthereumWalletService;
+import com.apollocurrency.aplwallet.apl.eth.utils.EthUtil;
+import com.apollocurrency.aplwallet.apl.exchange.model.DexCurrencies;
+import com.apollocurrency.aplwallet.apl.util.AplException;
+import com.apollocurrency.aplwallet.apl.util.Constants;
+import com.apollocurrency.aplwallet.apl.util.injectable.PropertiesHolder;
+import org.ethereum.util.blockchain.EtherUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.web3j.crypto.Credentials;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.tx.ClientTransactionManager;
+import org.web3j.tx.TransactionManager;
+import org.web3j.tx.gas.ContractGasProvider;
+import org.web3j.tx.gas.StaticGasProvider;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.concurrent.ExecutionException;
+
+import static com.apollocurrency.aplwallet.apl.util.Constants.ETH_DEFAULT_ADDRESS;
+
+@Singleton
+public class DexSmartContractService {
+    private static final Logger LOG = LoggerFactory.getLogger(DexSmartContractService.class);
+
+    private Web3j web3j;
+    private String smartContractAddress;
+    private String paxContractAddress;
+    private KeyStoreService keyStoreService;
+    private DexEthService dexEthService;
+    private EthereumWalletService ethereumWalletService;
+
+    @Inject
+    public DexSmartContractService(Web3j web3j, PropertiesHolder propertiesHolder, KeyStoreService keyStoreService, DexEthService dexEthService,
+                                   EthereumWalletService ethereumWalletService) {
+        this.web3j = web3j;
+        this.keyStoreService = keyStoreService;
+        smartContractAddress = propertiesHolder.getStringProperty("apl.eth.smart.contract.address");
+        paxContractAddress = propertiesHolder.getStringProperty("apl.eth.pax.contract.address");
+        this.dexEthService = dexEthService;
+        this.ethereumWalletService = ethereumWalletService;
+    }
+
+    /**
+     *  Deposit(freeze) money(eth or pax) on the contract.
+     * @param currency Eth or Pax
+     * @return String transaction hash.
+     */
+    public String deposit(String passphrase, long accountId, String fromAddress, BigInteger weiValue, Long gas, DexCurrencies currency) throws ExecutionException, AplException.ExecutiveProcessException {
+        WalletKeysInfo keyStore = keyStoreService.getWalletKeysInfo(passphrase, accountId);
+        EthWalletKey ethWalletKey = keyStore.getEthWalletForAddress(fromAddress);
```

Comment:
NPE is possible
keyStoreService.getWalletKeysInfo(passphrase, accountId)
can return NULL